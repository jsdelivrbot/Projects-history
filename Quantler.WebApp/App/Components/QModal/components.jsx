/*global module, require*/
var React = require("react");

module.exports = {

  Footer: React.createClass({
    render: function () {
      return (
        <div className="modal-footer">
             {this.props.children}
        </div>
      );
    }
  }),

  Body: React.createClass({
    render: function () {
      return (
        <div className='modal-body'>
             {this.props.children}
        </div>
      );
    }
  })

};
