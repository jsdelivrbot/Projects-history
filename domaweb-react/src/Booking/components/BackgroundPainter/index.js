import React from 'react';
import PropTypes from 'prop-types';

const randomColor = require('randomcolor');

class BackgroundPainter extends React.Component {
  // only repaint if something really changes
  shouldComponentUpdate(nextProps) {
    return nextProps.canvasTimeStart !== this.props.canvasTimeStart ||
           nextProps.canvasTimeEnd !== this.props.canvasTimeEnd ||
           nextProps.canvasWidth !== this.props.canvasWidth ||
           nextProps.activeItem !== this.props.activeItem ||
           nextProps.focusedItem !== this.props.focusedItem ||
           Object.values(nextProps.groupHeights).join(',') !== Object.values(this.props.groupHeights).join(',');
  }

  render() {
    const {
      groupTops,
      groupHeights,
      canvasWidth,
      groups,
    } = this.props;

    const backgrounds = [];
    for (let i = 0; i < groups.length; i++) {
      if (groups[i].id === this.props.activeItem || groups[i].id === this.props.focusedItem) {
        backgrounds.push(
          <div
            key={i}
            style={{
              top: groupTops[i],
              height: groupHeights[i],
              left: 0,
              width: canvasWidth,
              background: '#4D4D4D',
              position: 'absolute',
              zIndex: '50',
              opacity: '0.1',
            }}
          />,
        );
      } else {
        backgrounds.push(
          <div
            key={i}
            style={{
              top: groupTops[i],
              height: groupHeights[i],
              left: 0,
              width: canvasWidth,
              background: randomColor(),
            }}
          />,
        );
      }
    }
    return (
      <div style={{ display: 'absolute' }}>
        {backgrounds}
      </div>
    );
  }
}

export default BackgroundPainter;
