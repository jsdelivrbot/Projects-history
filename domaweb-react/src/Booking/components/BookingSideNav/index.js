/*
BookingSideNav container which wraps NavBookingTypes, NavMarkingTypes, NavMenu
 */

import React from 'react';
import PropTypes from 'prop-types';

import NavTimeTabs from '../../components/NavTimetabs';
import NavMenu from '../../components/NavMenu';
import NavBookingTypes from '../../components/NavBookingTypes';
import NavMarkingTypes from '../../components/NavMarkingTypes';

class BookingSideNav extends React.PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div>
        <NavMenu />
        <NavTimeTabs
          navitems={timetabs}
          onLinkClick={this.onLinkClicked.bind(this)}
          selectedIndex={this.props.params.timetabId}

        />
        <NavBookingTypes bookingtypes={bookingTypes} />
        <NavMarkingTypes />
      </div>
    );
  }
}

BookingSideNav.propTypes = {

};
export default BookingSideNav;
