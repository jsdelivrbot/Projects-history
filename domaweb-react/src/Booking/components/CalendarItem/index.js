import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { DragSource } from 'react-dnd';
import { connect as reduxConnect } from 'react-redux';
import { getEmptyImage } from 'react-dnd-html5-backend';
import { bindActionCreators } from 'redux';
import CalendarItem from './CalendarItem';
import { loadEmployeesGroup } from '../../containers/TimelineCalendar/actions';

const calendarSource = {
  beginDrag(props, monitor, component) {
    const { item } = props;
    return {
      item: {
        customerAddress: item.customerAddress,
        customerName: item.customerName,
        end: item.end,
        group: item.group,
        id: item.id,
        itemState: item.itemState,
        phoneNum: item.phoneNum,
        plannedEndTime: item.plannedEndTime,
        plannedStartTime: item.plannedStartTime,
        start: item.start,
        taskAddInfo: item.taskAddInfo,
        taskId: item.taskId,
        taskinfo: item.taskinfo,
        title: item.title,
        repetitionId: item.repetitionId,
      },
      type: props.type,
    };
  },
  canDrag(props, monitor) {
    return props.item.itemState !== 'FINISHED';
  },
  // isDragging(props, monitor) {
  //   const getClientOffset = monitor.getClientOffset();
  //   console.log('getInitialClientOffset client offset: ', getClientOffset);
  // },
}

function collect(connect, monitor) {
  return {
    connectDragSource: connect.dragSource(),
    connectDragPreview: connect.dragPreview(),
    isDragging: monitor.isDragging(),
  };
}

class DragCalendarItem extends Component {
  static propTypes = {
    connectDragSource: PropTypes.func.isRequired,
    connectDragPreview: PropTypes.func.isRequired,
    editTask: PropTypes.func.isRequired,
    item: PropTypes.shape({
      item: PropTypes.object,
      type: PropTypes.string,
    }),
    selectcalendaritem: PropTypes.func,
    selectedcalendaritem: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number,
    ]),
    isDragging: PropTypes.bool.isRequired,
  }

  componentDidMount() {
    this.props.connectDragPreview(getEmptyImage(), {
      captureDraggingState: true,
    });
  }

  // componentWillReceiveProps(nextProps) {
  //   if (this.props.isDragging !== nextProps.isDragging) {
  //     // this.props.setDragging(nextProps.isDragging);
  //     // console.log('nextProps.isDragging: ', nextProps.isDragging);
  //   }
  // }

  render() {
    const {
      connectDragSource,
      item,
      editTask,
      selectcalendaritem,
      selectedcalendaritem,
    } = this.props;

    return connectDragSource(
      <div>
        <CalendarItem
          item={item}
          editTask={editTask}
          selectcalendaritem={selectcalendaritem}
          selectedcalendaritem={selectedcalendaritem}
        />
      </div>,
    );
  }
}

const mapStateToProps = () => ({});

const mapDispatchToProps = dispatch => ({
  loadEmployeesGroup: bindActionCreators(loadEmployeesGroup, dispatch),
});

export default reduxConnect(mapStateToProps, mapDispatchToProps)(DragSource('CALENDAR', calendarSource, collect)(DragCalendarItem));
