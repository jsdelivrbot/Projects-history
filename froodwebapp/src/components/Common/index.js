// Common
export ColouredStatus from './ColouredStatus/ColouredStatus';
// AutoComplete
export AutoComplete from './AutoComplete/AutoComplete';
export PromotionAutoComplete from './AutoComplete/PromotionAutoComplete';
export ConnectedSkuAutoComplete from './AutoComplete/SkuAutoComplete';
export ConnectedBundleAutoComplete from './AutoComplete/BundleAutoComplete';
export ConnectedCustomerAutoComplete from './AutoComplete/CustomerAutoComplete';
export ConnectedCustomerOrderAutoComplete from './AutoComplete/CustomerOrderAutoComplete';
export ConnectedSupplierAutoComplete from './AutoComplete/SupplierAutoComplete';

// Buttons
export Button from './Buttons/Button/Button';
export ActionButton from './Buttons/ActionButton/ActionButton';
export TableActionButtons from './Buttons/TableActionButtons/TableActionButtons';
export Controls from './Controls/Controls';

// Modals
export ConnectedTopFormModal from './TopFormModal/TopFormModal';

// Tables
export ConnectedDynamicTable from './DynamicTable/DynamicTable';
export NewTableRow from './NewTableRow/NewTableRow';

// Header
export OrderTabHeader from './OrderTabHeader/OrderTabHeader';
export PageHeader from './PageHeader/PageHeader';

// Frood Antd Components
export FroodDatePicker from './FroodAntdComponents/FroodDatePicker';
export FroodInput from './FroodAntdComponents/FroodInput';
export FroodInputNumber from './FroodAntdComponents/FroodInputNumber';
export FroodSelect from './FroodAntdComponents/FroodSelect';
export FroodSwitch from './FroodAntdComponents/FroodSwitch';
export FroodTag from './FroodAntdComponents/FroodTag';
export FroodSearchInput from './FroodAntdComponents/FroodSearchInput';

// Grid Components
export GridTags from './GridComponents/GridTags/GridTags';
export GridInput from './GridComponents/GridInput/GridInput';
export GridInputNumber from './GridComponents/GridInputNumber/GridInputNumber';
export GridCheckbox from './GridComponents/GridCheckbox';
export GridTable from './GridComponents/GridTable/GridTable';
export ConnectedGridSkuAutoComplete from './GridComponents/GridSkuAutoComplete';
export GridButton from './GridComponents/GridButton/GridButton';
export GridSelect from './GridComponents/GridSelect/GridSelect';
export GridDatePicker from './GridComponents/GridDatePicker';

// Menu Link Item
export MenuLinkItem from './MenuLinkItem/MenuLinkItem';
