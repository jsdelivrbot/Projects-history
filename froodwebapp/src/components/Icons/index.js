export AddIcon from './AddIcon/AddIcon';
export CloseIcon from './CloseIcon/CloseIcon';
export OpenIcon from './OpenIcon/OpenIcon';
export DeleteIcon from './DeleteIcon/DeleteIcon';
export SortIcon from './SortIcon/SortIcon';
export RefreshIcon from './RefreshIcon/RefreshIcon';
export SettingsIcon from './SettingsIcon/SettingsIcon';
export FilterIcon from './FilterIcon/FilterIcon';
