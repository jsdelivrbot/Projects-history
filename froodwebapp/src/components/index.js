export Breadcrumb from './Breadcrumb/Breadcrumb';
export ConnectedHeader from './Header/Header';
export Footer from './Footer/Footer';
export * from './Common';

// Form
export FormGridTable from './ReduxFormFields/FormGridTable';
export FormTelephoneInput from './ReduxFormFields/FormTelephoneInput/FormTelephoneInput';
export FormRadioButtonGroup from './ReduxFormFields/FormRadioButtonGroup';
export FormRadioGroup from './ReduxFormFields/FormRadioGroup';
export FormRadio from './ReduxFormFields/FormRadio';
export FormDatePicker from './ReduxFormFields/FormDatePicker';
export FormCheckbox from './ReduxFormFields/FormCheckbox';
export FormSwitch from './ReduxFormFields/FormSwitch';
export FormSelect from './ReduxFormFields/FormSelect';
export FormInput from './ReduxFormFields/FormInput';
export FormTextArea from './ReduxFormFields/FormTextArea';
export FormInputNumber from './ReduxFormFields/FormInputNumber';
export FormTimePicker from './ReduxFormFields/FormTimePicker';
// Address Components
export FormAddressPopup from './ReduxFormFields/FormAddress/FormAddressPopup';
export FormAddressInfo from './ReduxFormFields/FormAddress/FormAddressInfo';
// Form AutoCompletes
export FormSkuAutoComplete from './ReduxFormFields/FormAutoComplete/FormSkuAutoComplete';
export FormBundleAutoComplete from './ReduxFormFields/FormAutoComplete/FormBundleAutoComplete';
export FormCustomerAutoComplete from './ReduxFormFields/FormAutoComplete/FormCustomerAutoComplete';
export FormCustomerOrderAutoComplete from './ReduxFormFields/FormAutoComplete/FormCustomerOrderAutoComplete';
export FormSupplierAutoComplete from './ReduxFormFields/FormAutoComplete/FormSupplierAutoComplete';
export FormPromotionAutoComplete from './ReduxFormFields/FormAutoComplete/FormPromotionAutoComplete';
