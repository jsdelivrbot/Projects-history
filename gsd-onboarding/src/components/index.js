export { SideNav } from './CommonBlocks/SideNav/SideNav';
export { NavButtons } from './CommonBlocks/NavButtons/NavButtons';
export { NextButton } from './CommonBlocks/NavButtons/NextButton';
export { WelcomePage } from './WelcomePage/WelcomePage';
