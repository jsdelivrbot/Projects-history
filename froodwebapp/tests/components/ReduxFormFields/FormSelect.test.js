import React from 'react';
import { FormSelect } from 'components';
import renderer from 'react-test-renderer';

const { describe, it, expect } = global;

describe('FormSelect', () => {
  it('captures snapshot of FormSelect', () => {
    const props = {
      input: {
        value: 'properties',
        onChange: jest.fn()
      },
      menuItems: [{
        key: 1,
        value: 'value1'
      }, {
        key: 2,
        value: 'value2'
      }],
      disabled: true,
      placeholder: 'Enter value'
    };
    const snappedComponent = renderer.create(<FormSelect { ...props } />).toJSON();

    expect(snappedComponent).toMatchSnapshot();
  });
});
