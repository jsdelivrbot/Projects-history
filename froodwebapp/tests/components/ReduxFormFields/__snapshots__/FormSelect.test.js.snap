// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormSelect captures snapshot of FormSelect 1`] = `
<div
  className="select ant-select ant-select-disabled"
  onBlur={[Function]}
  onClick={undefined}
  onFocus={[Function]}
  onMouseDown={undefined}
  onMouseEnter={undefined}
  onMouseLeave={undefined}
  onTouchStart={undefined}
  style={undefined}
>
  <div
    aria-autocomplete="list"
    aria-expanded={false}
    aria-haspopup="true"
    className="ant-select-selection
            ant-select-selection--single"
    onKeyDown={[Function]}
    role="combobox"
    tabIndex={0}
  >
    <div
      className="ant-select-selection__rendered"
    >
      <div
        className="ant-select-selection__placeholder"
        onClick={[Function]}
        onMouseDown={[Function]}
        style={
          Object {
            "WebkitUserSelect": "none",
            "display": "none",
            "userSelect": "none",
          }
        }
        unselectable="unselectable"
      >
        Enter value
      </div>
      <div
        className="ant-select-selection-selected-value"
        style={
          Object {
            "display": "block",
            "opacity": 1,
          }
        }
        title="properties"
      >
        properties
      </div>
      <div
        className="ant-select-search ant-select-search--inline"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <div
          className="ant-select-search__field__wrap"
        >
          <input
            autoComplete="off"
            className="ant-select-search__field"
            disabled={true}
            id={undefined}
            onChange={[Function]}
            onKeyDown={[Function]}
            value=""
          />
          <span
            className="ant-select-search__field__mirror"
          >
            
            Â 
          </span>
        </div>
      </div>
    </div>
    <span
      className="ant-select-arrow"
      onClick={[Function]}
      style={
        Object {
          "WebkitUserSelect": "none",
          "userSelect": "none",
        }
      }
      unselectable="unselectable"
    >
      <b />
    </span>
  </div>
</div>
`;
